### PR TITLE
Session.model.test

### PR DESCRIPTION
**Pull request by matthewphilyaw**
Should cover the entire session model now.

### CODE REVIEW

**Comment by matthewphilyaw** [comment]
I think there is one more test needed around if a username key is provided, I do one to check for nil but I think it will blow up the username is not provided. Will look at this later tonight.

**Comment by copenhas** [comment]
This one might need to be rebased. It's trying to add 3 commits and one of those I think was already merged into master.

**Comment by matthewphilyaw** [comment]
Thought I rebased it when I get home I'll ok at it and the other issue you were having. 

Actually I bet that is the one where I pushed origin by accident. I'll sort it out sorry about that.

File: test/models/session_test.exs [hunk 1]
```diff
@@ -6,6 +6,27 @@ defmodule Yggdrasil.SessionTest do
...
+
+  defp setup_user() do
+    user = %{
+      :username => @correct_username,
+      :password => @correct_password,
+      :password_confirmation => @correct_password
+    }
+
+    User.create_changeset(%User{}, user)
+      |> Repo.insert()
```
Comment (copenhas) [inline test/models/session_test.exs:23]: So does ecto do a rollback after each test? Or are we populating the database and this will fail when you run it two times in a row?

**Comment by matthewphilyaw** [comment]
https://github.com/voluspa/yggdrasil/blob/86f67b60228cc6a098a8cece9c7a013159063831/test/test_helper.exs

Starts a test transaction, I run these constantly and no users are ever added. There is a another example of this in the support/conn_case.exs file as well I think it does something with a transaction as long as the tests aren't in parallel, the async tag isn't used.

**Comment by copenhas** [comment]
Sounds good to me.
